docs/build_and_push_docker_image_workflow_documentation.md

```diff
@@ -0,0 +1,219 @@
+# Build and Push Docker Image Workflow Documentation
```

Comment:
When I said that I wanted it documented, I meant a write up in the issue or PR so I could understand the expectations behind this.  Unsure how much of this will be useful in a living document.  If there is, it likely more belongs in `CONTRIBUTING.md`

Reply:
Based on the understanding from https://github.com/crate-ci/typos/pull/1116#discussion_r1823323694, I don't think we need to keep this documentation.